bst_plugin/mbll/process_nst_dOD.m

```diff
@@ -149,35 +151,61 @@
      if strcmp(cond_name(1:4), '@raw')
         cond_name = cond_name(5:end);
      end
-     iStudy = db_add_condition(sInputs.SubjectName, [cond_name, '_dOD']);
+
+     if isRaw
+        iStudy = db_add_condition(sInputs.SubjectName, ['@raw', cond_name, '_dOD']);
+     else
+         iStudy = db_add_condition(sInputs.SubjectName, [cond_name, '_dOD']);
+     end
      sStudy = bst_get('Study', iStudy);
-%     
+     
 %     % Save channel definition
      [tmp, iChannelStudy] = bst_get('ChannelForStudy', iStudy);
      db_set_channel(iChannelStudy, ChannelMat, 0, 0);
         
-    % Save time-series data
-%     final_nirs = sDataIn.F;
-%     final_nirs(to_keep, :) = final_nirs';
-    sDataOut = db_template('data');
-    sDataOut.F            = final_dOD';
-    sDataOut.Comment      = sDataIn.Comment;
-    %sDataOut.ChannelFlag  = sDataIn.ChannelFlag;
-    sDataOut.ChannelFlag  = ones(size(final_dOD, 2), 1);
-    sDataOut.Time         = sDataIn.Time;
-    sDataOut.DataType     = 'recordings'; 
-    sDataOut.nAvg         = 1;
-    sDataOut.Events       = events;
-    sDataOut.History      = sDataIn.History;
-    sDataOut              = bst_history('add', sDataOut, 'process', sProcess.Comment);
-    sDataOut.DisplayUnits = 'delta OD';
-
-    % Generate a new file name in the same folder
-    OutputFile = bst_process('GetNewFilename', bst_fileparts(sStudy.FileName), 'data_OD');
-    sDataOut.FileName = file_short(OutputFile);
-    bst_save(OutputFile, sDataOut, 'v7');
-    % Register in database
-    db_add_data(iStudy, OutputFile, sDataOut);
+     if ~isRaw
+        % Save time-series data
+        sDataOut = db_template('data');
+        sDataOut.F            = final_dOD';
+        sDataOut.Comment      = sDataIn.Comment;
+        %sDataOut.ChannelFlag  = sDataIn.ChannelFlag;
+        sDataOut.ChannelFlag  = ones(size(final_dOD, 2), 1);
+        sDataOut.Time         = sDataIn.Time;
+        sDataOut.DataType     = 'recordings'; 
+        sDataOut.nAvg          = 1;
+        sDataOut.Events       = events;
+        sDataOut.History      = sDataIn.History;
+        sDataOut              = bst_history('add', sDataOut, 'process', sProcess.Comment);
+        sDataOut.DisplayUnits = 'delta OD';
+        
+        sDataOut.FileName = file_short(OutputFile);
```

Comment:
Missing line before:
`OutputFile = bst_process('GetNewFilename', bst_fileparts(sStudy.FileName), 'data_OD');`

It could be added before the IF to work for the `raw` and `non-raw`

Reply:
done